packages/cli-utils/src/index.ts

```diff
@@ -155,8 +138,95 @@ async function main() {
       const shouldPreprocess =
         !options['disable-preprocessing'] && options['disable-preprocessing'] !== 'false';
       return generateTadaTypes(shouldPreprocess);
+    })
+    .command('check')
+    .option('--severity', 'The minimum severity of diagnostics to display.')
```

Comment:
(doesn't block merging, can be a todo)

As per the other comment, I'd suggest a `--level` but let's maybe also add a `--skip-warnings` or `--fail-on-warn` and by default not fail on warnings (and similarly to info, hide detailed outputs when we succeed / exit with code `0`)